akka-stream/src/main/scala/akka/stream/impl/FanIn.scala

```diff
@@ -183,13 +204,19 @@ private[akka] object FanIn {
     }
 
     def inputsAvailableFor(id: Int) = new TransferState {
-      override def isCompleted: Boolean = depleted(id) || cancelled(id)
-      override def isReady: Boolean = pending(id)
+      override def isCompleted: Boolean = {
+        val state = states(id)
+        has(state, Depleted) || has(state, Cancelled)
+      }
```

Comment:
Yes, but it also creates more opportunity for error, e.g. in `idToDequeue` above. What about adding `hasAnyOf` and `hasAllOf` methods?